### DISCUSSION TITLE
[Thaumato] Initial blurring

### DISCUSSION THREAD

**Discussion by giorgioangel** [ScrollPrize/villa#39] [General]
The very first step of the pipeline is blurring the volume to stabilize the subsequent gradient estimation via Sobel filters with the final aim of surface (edge) detection. 

Blurring is performed with a uniform box filter of kernel $11x11x11$ (which is almost the thickness of a papyrus sheet with voxel size $7.91 \mu m$). 

The operations are performed here:
https://github.com/ScrollPrize/villa/blob/d4b0113b2dfe0edc6d882dd33a8b341625bc6ae7/thaumato-anakalyptor/ThaumatoAnakalyptor/grid_to_pointcloud.py#L596 ,
https://github.com/ScrollPrize/villa/blob/d4b0113b2dfe0edc6d882dd33a8b341625bc6ae7/thaumato-anakalyptor/ThaumatoAnakalyptor/surface_detection.py#L287 and https://github.com/ScrollPrize/villa/blob/d4b0113b2dfe0edc6d882dd33a8b341625bc6ae7/thaumato-anakalyptor/ThaumatoAnakalyptor/surface_detection.py#L91

The use of a uniform box filter is computationally efficient, however, I wonder whether this step of the pipeline could potentially benefit from optimization, especially regarding adopting a different denoising scheme and kernel size.

I cite [Jain and Tyagi (2014)](https://link.springer.com/article/10.1007/s10796-014-9527-0):
>Ideally denoising is all about filtering noise from the degraded image while keeping other details unchanged. Indeed, filtering is the most fundamental operation of image processing and computer vision, and it is used extensively in a wide range of applications, including image smoothing and sharpening, noise removal, resolution enhancement and reduction, feature extraction, and edge detection. In the broadest sense of the term “filtering,” the value of the filtered image at a given location is a function of the values of the input image in a small neighborhood of the same location (Jain and Tyagi [2013](https://link.springer.com/article/10.1007/s10796-014-9527-0#ref-CR37)).
> The simplest filtering should be explicit linear translation invariant (LTI) filtering, which can be implemented using a convolution mask. For example, box filtering, also known as mean filtering or smoothing or averaging (Gonzalez and Woods [2006](https://link.springer.com/article/10.1007/s10796-014-9527-0#ref-CR31)), is implemented by a local averaging operation where the value of each pixel is replaced by the average of all the values in the local neighborhood. Box filter is the quickest blur algorithm, but its smoothing effect is often not sufficient.

and also

> Linear filters like mean filter (Gonzalez and Woods [2006](https://link.springer.com/article/10.1007/s10796-014-9527-0#ref-CR31)), Wiener filter (Jain [1989](https://link.springer.com/article/10.1007/s10796-014-9527-0#ref-CR36); Benesty etal. [2010](https://link.springer.com/article/10.1007/s10796-014-9527-0#ref-CR7)) too tend to blur sharp edges, destroy lines and other fine image details, and perform poorly in the presence of signal-dependent noise.
> Mean filtering is a simple, intuitive and easy to implement method of reducing noise in images. The idea of mean filtering is simply to replace each pixel value in an image with the mean (‘average’) value of its neighbors, including itself.

Given the signal-dependent nature of Poisson noise in X-ray CT scans, it may be worth exploring whether alternative denoising techniques, such as [BM4D](https://webpages.tuni.fi/foi/papers/bm4d_preprint.pdf) or [Non-Local Means](https://www.ipol.im/pub/art/2011/bcm_nlm/article.pdf), could further enhance edge detection accuracy while preserving fine details.

**Comment by ubc-nvining** [comment]
I've been advocating this to @hariseldon137 for awhile. I'm not against denoising per se but an 11^3 seems like it's not going to do good things to the data for subsequent steps. I can't really comment intelligently on the physical implications of doing an 11^3 blur but just looking at the voxel data alone, from a geometry perspective, I have to imagine that it is eliminating useful information and probably making some elements worse, especially in highly compressed regions.
